cudamapper/CMakeLists.txt

```diff
@@ -13,6 +13,22 @@ project(cudamapper)
 # Process data subdirectory first
 add_subdirectory(data)
 
+execute_process(COMMAND
+  git describe --tag --dirty
+  WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+  OUTPUT_VARIABLE VERSION
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+string(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" VERSION_MAJOR "${VERSION}")
+string(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" VERSION_MINOR "${VERSION}")
+string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" VERSION_PATCH "${VERSION}")
+string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.[0-9]-+(.*)" "\\1" VERSION_SHA1 "${VERSION}")
+set(VERSION_SHORT "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/templates/version.cpp.in
+                ${CMAKE_CURRENT_BINARY_DIR}/version.cpp)
+set(version_file "${CMAKE_CURRENT_BINARY_DIR}/version.cpp")
```

Comment:
we can probably just do with generating a configured `.hpp` file without needing a `.cpp` file

Reply:
v0.4.4-505-g4524e41-dirty

In the version string above "-dirty" is added if there is anything uncommitted in the local repo or if a version controlled file is edited. If we edit version.hpp during the build, any subsequent git command to find the version will include '-dirty' in the version string. It will not hurt this component, but can become a problem for others in CI pipeline.

Here version.cpp is generated and not version controlled. So we circumvent this issue.

Reply:
I think it makes sense to have `version.hpp` and `version.cpp` for better separation. This way projects won't have to recompile if we make a minor version update.

Is there a benefit of using a `version_file` variable? Why don't you just add the `${CMAKE_CURRENT_BINARY_DIR}/version.cpp` to the `cuda_add_library` call directly?